include/openthread/types.h

```diff
@@ -683,6 +683,8 @@ enum
     OT_CHANGED_THREAD_CHILD_REMOVED         = 1 << 11,  ///< Child was removed
     OT_CHANGED_IP6_MULTICAST_SUBSRCRIBED    = 1 << 12,  ///< Subscribed to a IPv6 multicast address
     OT_CHANGED_IP6_MULTICAST_UNSUBSRCRIBED  = 1 << 13,  ///< Unsubscribed from a IPv6 multicast address
+    OT_CPU_POWER_HI                         = 1 << 14,  ///< Hi CPU power is needed
```

Comment:
In general, I believe OpenThread should provide mechanisms but not policy wherever possible.  As @DuaneEllis-TI mentioned, there are a number of approaches to optimizing performance.  Simply stating that the platform should enter "high power" or "performance needed" state whenever the Commissioner/Joiner roles are enabled is a specification of policy.  What if another platform has a more fine-grained approach to optimizing performance of crypto operations?

I would prefer to provide notifications in changes to the Commissioner/Joiner state.  Then users could do what they want with that notification, some of which may not be related to optimizing crypto performance at all.